src/Repository/AbstractRepository.php

```diff
@@ -323,16 +328,13 @@ public function getFields(EntityInterface $entity)
      */
     public function getEntitiesFromTarget(EntityInterface $entity, EntityInterface $target): array
     {
-        $entityId = $entity->getId();
-        $targetId = $target->getId();
+
         $entityTable = $entity->getTableName();
         $targetTable = $target->getTableName();
         $fields = $this->getFields($entity);
 
         $query = 'SELECT ' . $fields . ' FROM ' . $entityTable . ' INNER JOIN ' . $targetTable . ' ON ' . $targetTable . '.id = ' . $entityTable . '.' . $targetTable . '_id;';
```

Comment:
think about making a separate class (e.g. SQLBuilder/DataAccess) that has  the responsibility of building complex SQL clauses. If you want, you can make a separate issue for this.